tests/app/data/test_schema_validation.py

```diff
@@ -46,6 +46,35 @@ def test_schema(self):
 
             self.fail("{} Schema Validation Errors.".format(len(errors)))
 
+    def test_no_duplicate_ids_in_schema(self):
+        schema_files = schema_loader.available_schemas()
+        ignored_keys = ['routing_rules', 'skip_condition']
+
+        print('Checking schema files for duplicate keys.')
+        for schema_file in schema_files:
+            unique_id = []
+            with open(os.path.join(settings.EQ_SCHEMA_DIRECTORY, schema_file), encoding="utf8") as file:
+                schema_json = json.load(file)
+                print('Checking %s...' % schema_file)
+                for id_value in self._parse_id_values(schema_json, ignored_keys):
+                    if id_value in unique_id:
+                        self.fail('Duplicate Id found. schema: %s, id: %s' % (schema_file, id_value))
+                    else:
+                        unique_id.append(id_value)
+
+    def _parse_id_values(self, schema_json, ignored_keys):
+        for k, v in schema_json.items():
+            if k == 'id':
+                yield schema_json[k]
+            elif k in ignored_keys:
```

Comment:
Oh yeah doh! Might would worth giving them a name to reflect that. That's a tough ask so might be worth a comment to state they contain id references which are likely to contain duplicates